storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -181,13 +187,18 @@ func (s *commitLogSource) ReadData(
 	wg.Wait()
 	s.logEncodingOutcome(workerErrs, iter)
 
-	return s.mergeShards(int(numShards), bopts, blockSize, blopts, encoderPool, unmerged), nil
+	result := s.mergeShards(int(numShards), bopts, blockSize, blopts, encoderPool, unmerged)
+	// After merging shards, its safe to cache the shardData (which involves some mutation).
+	if s.opts.CacheSeriesMetadata() {
```

Comment:
Hm instead of making it an option, its's probably best to infer this from `nsMetadata.Options().IndexOptions().Enabled()`.  Otherwise its an extra flip that you pretty much always want to use.

Reply:
Thats a good idea, I'll add that as an additional check, but I think the separate config is still good in-case its using too much memory, or we want to just debug this feature on/off.

Also you just made me realize I should probably segregate the cached data by namespace...

Reply:
done and done